model-optimizer/extensions/ops/elementwise.py

```diff
@@ -22,13 +22,19 @@ def override_data_type_of_constant(node: Node):
         # power value 2, or when replacing Neg operation with Mul with -1 as second input.
         in_node_0 = node.in_port(0).get_source().node
         in_node_1 = node.in_port(1).get_source().node
+
+        if in_node_0.op != 'Const' and in_node_1.op != 'Const':
+            raise Exception("Elementwise operation '{}' has inputs of different data types: '{}' and '{}' "
+                            "that cannot be aligned".format(node.soft_get('name'), in_type_0, in_type_1))
+
         if in_node_0.op == 'Const':
-            convert_const_node_value_type(in_node_0, in_type_1)
-        elif in_node_1.op == 'Const':
-            convert_const_node_value_type(in_node_1, in_type_0)
+            node_to_convert, src_type, dst_type = in_node_0, in_type_0, in_type_1
         else:
-            log.error('Elementwise operation {} has inputs of different data types: {} and {}'.format(
-                node.soft_get('name'), in_type_0, in_type_1))
+            node_to_convert, src_type, dst_type = in_node_1, in_type_1, in_type_0
+        log.error("Changing Const node '{}' data type from {} to {} for Elementwise operation".format(
+            node.soft_get('name', node_to_convert.id), src_type, dst_type),
```

Comment:
Should it be `node_to_convert.soft_get(...` instead of `node.soft_get(...`?

Reply:
fixed